specification/app/resource-manager/Microsoft.App/stable/2025-07-01/ConnectedEnvironmentsStorages.json

```diff
@@ -0,0 +1,330 @@
+            "in": "path",
+            "description": "Name of the Environment.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "storageName",
+            "in": "path",
+            "description": "Name of the storage.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Delete operation is accepted",
+            "headers": {
+              "Location": {
+                "type": "string"
+              }
+            }
+          },
+          "204": {
+            "description": "Storage does not exist"
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List environments storages by subscription": {
+            "$ref": "./examples/ConnectedEnvironmentsStorages_Delete.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ConnectedEnvironmentStoragesCollection": {
+      "description": "Collection of Storage for Environments",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of storage resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConnectedEnvironmentStorage"
+          }
+        }
+      }
+    },
+    "ConnectedEnvironmentStorage": {
+      "description": "Storage resource for connectedEnvironment.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Storage properties",
+          "type": "object",
+          "properties": {
+            "azureFile": {
+              "$ref": "./CommonDefinitions.json#/definitions/AzureFileProperties",
+              "description": "Azure file properties"
+            },
+            "provisioningState": {
+              "description": "Provisioning state of the storage.",
+              "enum": [
+                "Succeeded",
+                "Failed",
+                "Canceled",
+                "InProgress",
+                "Deleting"
+              ],
+              "type": "string",
+              "readOnly": true,
+              "x-ms-enum": {
+                "name": "ConnectedEnvironmentStorageProvisioningState",
+                "modelAsString": true
+              }
+            },
+            "deploymentErrors": {
```

Comment:
>deploymentErrors [](http://example.com/codeflow?start=13&length=16)

Make name/description singular or change to array for collection.

Reply:
It's stable release since https://github.com/Azure/azure-rest-api-specs/blob/6c2a7889cad5b66015f6d7448cc07179039ae8b1/specification/app/resource-manager/Microsoft.App/stable/2022-10-01/ConnectedEnvironments.json#L393

Shall we avoid breaking changes？

Reply:
If its defined already in ConnectedEnvironments. why not refer from there , instead of creating a new one in multiple places?

Reply:
It's different API versions, under folder name 2025-01-01 or 2022-10-01, each folder for each version as required.

Reply:
ConnectedEnvironments and ConnectedEnvironmentsStorages are different resource type, but they shared the same service API pattern, we have deploymentErrors as string type in many resources since very beginning.